editoast/src/views/path/path_item_cache.rs

```diff
@@ -408,3 +461,136 @@ fn secondary_code_filter(
+                reference: OperationalPointIdentifier::OperationalPointDescription {
+                    trigram: NonBlankString::from("DEF"),
+                    secondary_code: None,
+                },
+                track_reference: None,
+            }),
+            PathItemLocation::OperationalPointReference(OperationalPointReference {
+                reference: OperationalPointIdentifier::OperationalPointUic {
+                    uic: 5678,
+                    secondary_code: None,
+                },
+                track_reference: None,
+            }),
+        ];
+
+        let path_item_refs: Vec<&PathItemLocation> = path_items.iter().collect();
+
+        // Load the cache using the real load() method
+        let cache = PathItemCache::load(&mut conn, infra.id, &path_item_refs)
+            .await
+            .expect("Failed to load cache");
+
+        // Test OP1 (has both trigram and UIC) can be found by all methods
+        assert!(
+            cache.get_from_id("op_1").is_some(),
+            "OP1 should be findable by ID"
+        );
+        assert!(
+            cache.get_from_trigram("ABC").is_some(),
+            "OP1 should be findable by trigram even though queried by ID"
+        );
+        assert!(
+            cache.get_from_uic(1234).is_some(),
+            "OP1 should be findable by UIC even though queried by ID"
+        );
+
+        // Verify it's the same OP
+        let op_by_id = cache.get_from_id("op_1").unwrap();
+        let op_by_trigram = cache.get_from_trigram("ABC").unwrap().first().unwrap();
+        let op_by_uic = cache.get_from_uic(1234).unwrap().first().unwrap();
+        assert_eq!(op_by_id.obj_id, "op_1");
+        assert_eq!(op_by_trigram.obj_id, "op_1");
+        assert_eq!(op_by_uic.obj_id, "op_1");
+
+        // Test OP2 (trigram only) is indexed by ID and trigram but not UIC
+        assert!(
+            cache.get_from_id("op_2").is_some(),
+            "OP2 should be findable by ID even though queried by trigram"
+        );
+        assert!(
+            cache.get_from_trigram("DEF").is_some(),
+            "OP2 should be findable by trigram"
+        );
+        let op2_by_id = cache.get_from_id("op_2").unwrap();
+        let op2_by_trigram = cache.get_from_trigram("DEF").unwrap().first().unwrap();
+        assert_eq!(op2_by_id.obj_id, op2_by_trigram.obj_id);
+
+        // Test OP3 (UIC only) is indexed by ID and UIC but not trigram
+        assert!(
+            cache.get_from_id("op_3").is_some(),
+            "OP3 should be findable by ID even though queried by UIC"
+        );
+        assert!(
+            cache.get_from_uic(5678).is_some(),
+            "OP3 should be findable by UIC"
+        );
+        let op3_by_id = cache.get_from_id("op_3").unwrap();
+        let op3_by_uic = cache.get_from_uic(5678).unwrap().first().unwrap();
+        assert_eq!(op3_by_id.obj_id, op3_by_uic.obj_id);
```

Comment:
I find the following more explicit on what we are suppose to test: we do not test equality, we test the output value (equality is only a side effect).

```suggestion
        assert_eq!(cache.get_from_id("op_3").unwrap().obj_id, "op_3");
        assert_eq!(cache.get_from_uic(5678).unwrap().first().unwrap().obj_id, "op_3");
```

Reply:
Done in [8b1448d6](https://github.com/OpenRailAssociation/osrd/compare/8b1448d64c27731e6b0137d1a158bad48118a931..d26c8fa7b335a3d97781ee3edb1beb73fbe52666)